tests/integration/lldp_test.py

```diff
@@ -0,0 +1,319 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+from contextlib import contextmanager
+import time
+import yaml
+
+import libnmstate
+from libnmstate.schema import Interface
+from libnmstate.schema import LLDP
+
+from .testlib import assertlib
+from .testlib import cmdlib
+from .testlib import statelib
+
+
+ETH1 = "eth1"
+
+LLDP_SYSTEM_DESC = (
+    "Summit300-48 - Version 7.4e.1 (Build 5) by Release_Master "
+    "05/27/05 04:53:11"
+)
+
+EXPECTED_LLDP_NEIGHBOR = """
+- system-name: Summit300-48
+  type: 5
+- system-description: Summit300-48 - Version 7.4e.1 (Build 5) by Release_Master
+    05/27/05 04:53:11
+  type: 6
+- system-capabilities:
+  - MAC Bridge component
+  - Router
+  type: 7
+- _description: MAC address
+  chassis-id: 00:01:30:F9:AD:A0
+  chassis-id-type: 4
+  type: 1
+- _description: Interface name
+  port-id: 1/1
+  port-id-type: 5
+  type: 2
+- ieee-802-1-vlans:
+  - name: v2-0488-03-0505
+    vid: 488
+  oui: 00:80:c2
+  subtype: 3
+  type: 127
+- ieee-802-3-mac-phy-conf:
+    autoneg: true
+    operational-mau-type: 16
+    pmd-autoneg-cap: 27648
+  oui: 00:12:0f
+  subtype: 1
+  type: 127
+- ieee-802-1-ppvids:
+  - 0
+  oui: 00:80:c2
+  subtype: 2
+  type: 127
+- management-addresses:
+  - address: 00:01:30:F9:AD:A0
+    address-subtype: MAC
+    interface-number: 1001
+    interface-number-subtype: 2
+  type: 8
+- ieee-802-3-max-frame-size: 1522
+  oui: 00:12:0f
+  subtype: 4
+  type: 127
+"""
+
+LLDP_CAPS = ["MAC Bridge component", "Router"]
+
+CHASSIS_ID = "chassis-id"
+CHASSIS_ID_TYPE = "chassis-id-type"
+MANAGEMENT_ADDRESSES = "management-addresses"
+ADDRESS = "address"
+ADDRESS_SUBTYPE = "address-subtype"
+INTERFACE_NUMBER = "interface-number"
+INTERFACE_NUMBER_SUBTYPE = "interface-number-subtype"
+AUTONEG = "autoneg"
+OPERATIONAL_MAU_TYPE = "operational-mau-type"
+PMD_AUTONEG_CAP = "pmd-autoneg-cap"
+PPVIDS_SUBTREE = "ieee-802-1-ppvids"
+PPVID = "ppvid"
+VLANS_SUBTREE = "ieee-802-1-vlans"
+NAME = "name"
+PORT_ID = "port-id"
+PORT_ID_TYPE = "port-id-type"
+SYSTEM_NAME = "system-name"
+SYSTEM_DESCRIPTION = "system-description"
+SYSTEM_CAPABILITIES = "system-capabilities"
+MFS_KEY = "ieee-802-3-max-frame-size"
+MAC_PHY_SUBTREE = "ieee-802-3-mac-phy-conf"
+
+
+def test_enable_lldp(eth1_up):
+    with lldp_enabled(eth1_up) as dstate:
+        assertlib.assert_state_match(dstate)
+
+
+def test_lldp_yaml(eth1_up):
+    with lldp_enabled(eth1_up):
+        _send_lldp_packet()
+        dstate = statelib.show_only((ETH1,))
+        lldp_config = dstate[Interface.KEY][0][LLDP.CONFIG_SUBTREE]
+        test_neighbor = _get_lldp_test(lldp_config[LLDP.NEIGHBORS_SUBTREE])
```

Comment:
We cannot assume that there is no more neighbors that the one we are injecting. When testing LLDP in different platforms I have found that we got more that one if there a switch or something providing LLDP information.

This is only true when using our container.

Reply:
tcpreplay is injecting on eth1peer, which implies that it is a veth interface. Maybe the test should then setup its custom veth interfaces that it would also use on other platforms to avoid conflicts when eth1 is actuall not a veth.

Reply:
All of the platforms of NMCI is using veth interfaces, I like this idea but could we do it in another PR? That would require specific testing in platforms with not veths. Thanks!

Reply:
You wrote that a switch might inject LLDP information on different platforms but you also state that veths are used on every platform but the switches should not be able to inject LLDP information in the veth. Therefore, it seems to me that you can just add the assertion for one neighbor. If I am missing something and it is too complicated, I am fine with revising the test. Please track this in an issue, then.

Reply:
I will add the test checking that there is only one neighbor. You are right in your statement, thanks!

Reply:
Done, I've added a test. Thanks!

Reply:
All the `len()` asserts are happening after a filter AFAICS, so there is no assertion to check for only one neighbor.